### ISSUE TITLE
Deploy Azure App Service Error - Error: Endpoint auth data not present

### ISSUE THREAD

**Issue by andrewlock** [Microsoft/vsts-tasks#8442]
## Environment
- Server - Azure Pipelines
        - If using Azure Pipelines, provide the account name, team project name, build definition name/build number: 
  - Account Name: andrewlocknet
  - Project name: BlogFunctions
  - Release: BlogFunctions release pipeline: Release-3

- Agent - Hosted: 
    - If using Hosted agent, provide agent queue name: 
      - vs2017-win2017

## Issue Description

I've created a simple release that should deploy the master branch to production. It's a functions app, where the output is a zip file. I used the "Deploy Azure App Service v4 (preview)" task and linked my Azure description. There appears to be an authentication issue as I am seeing the error `Error: Endpoint auth data not present: 424ebb0f-11b3-45ba-a37e-9bddb5ee917c`. It looks like the subscription information is not being picked up, though I can see it listed in "Project Settings > Service connections". 

It looks to be the same issue as https://github.com/Microsoft/vsts-tasks/issues/8308 and [this](https://developercommunity.visualstudio.com/content/problem/324154/error-endpoint-auth-data-not-present.html) in which the solution was "Could you manually add the helm task for the time being?", but I don't know what that means (I'm new to Azure DevOps and Azure tbh).

### Task logs

```
2018-09-28T06:03:04.5810679Z ##[debug]Evaluating condition for step: 'Deploy Azure App Service'
2018-09-28T06:03:04.5811578Z ##[debug]Evaluating: succeeded()
2018-09-28T06:03:04.5811715Z ##[debug]Evaluating succeeded:
2018-09-28T06:03:04.5811904Z ##[debug]=> True
2018-09-28T06:03:04.5812123Z ##[debug]Result: True
2018-09-28T06:03:04.5812369Z ##[section]Starting: Deploy Azure App Service
2018-09-28T06:03:04.5819200Z ==============================================================================
2018-09-28T06:03:04.5819269Z Task         : Azure App Service Deploy
2018-09-28T06:03:04.5819343Z Description  : Update Azure App Services on Windows, Web App on Linux with built-in images or Docker containers, ASP.NET, .NET Core, PHP, Python or Node.js based Web applications, Function Apps on Windows or Linux with Docker Containers, Mobile Apps, API applications, Web Jobs using Web Deploy / Kudu REST APIs
2018-09-28T06:03:04.5819782Z Version      : 4.3.6
2018-09-28T06:03:04.5819833Z Author       : Microsoft Corporation
2018-09-28T06:03:04.5819906Z Help         : [More information](https://aka.ms/azurermwebdeployreadme)
2018-09-28T06:03:04.5819953Z ==============================================================================
2018-09-28T06:03:04.8695295Z ##[debug]agent.workFolder=D:\a
2018-09-28T06:03:04.8747409Z ##[debug]loading inputs and endpoints
2018-09-28T06:03:04.8754656Z ##[debug]loading ENDPOINT_AUTH_PARAMETER_SYSTEMVSSCONNECTION_ACCESSTOKEN
2018-09-28T06:03:04.8765044Z ##[debug]loading ENDPOINT_AUTH_SCHEME_SYSTEMVSSCONNECTION
2018-09-28T06:03:04.8768006Z ##[debug]loading ENDPOINT_AUTH_SYSTEMVSSCONNECTION
2018-09-28T06:03:04.8771265Z ##[debug]loading INPUT_CONNECTEDSERVICENAME
2018-09-28T06:03:04.8772875Z ##[debug]loading INPUT_CONNECTIONTYPE
2018-09-28T06:03:04.8776953Z ##[debug]loading INPUT_DEPLOYMENTTYPE
2018-09-28T06:03:04.8805448Z ##[debug]loading INPUT_DEPLOYTOSLOTORASEFLAG
2018-09-28T06:03:04.8806459Z ##[debug]loading INPUT_EXCLUDEFILESFROMAPPDATAFLAG
2018-09-28T06:03:04.8807391Z ##[debug]loading INPUT_PACKAGE
2018-09-28T06:03:04.8807630Z ##[debug]loading INPUT_PUBLISHPROFILEPATH
2018-09-28T06:03:04.8807837Z ##[debug]loading INPUT_REMOVEADDITIONALFILESFLAG
2018-09-28T06:03:04.8808395Z ##[debug]loading INPUT_RENAMEFILESFLAG
2018-09-28T06:03:04.8808733Z ##[debug]loading INPUT_SCRIPTPATH
2018-09-28T06:03:04.8809090Z ##[debug]loading INPUT_SETPARAMETERSFILE
2018-09-28T06:03:04.8809821Z ##[debug]loading INPUT_SLOTNAME
2018-09-28T06:03:04.8819096Z ##[debug]loading INPUT_TAKEAPPOFFLINEFLAG
2018-09-28T06:03:04.8819793Z ##[debug]loading INPUT_USEWEBDEPLOY
2018-09-28T06:03:04.8820895Z ##[debug]loading INPUT_WEBAPPKIND
2018-09-28T06:03:04.8821564Z ##[debug]loading INPUT_WEBAPPNAME
2018-09-28T06:03:04.8829726Z ##[debug]loading INPUT_XMLTRANSFORMATION
2018-09-28T06:03:04.8831162Z ##[debug]loading INPUT_XMLVARIABLESUBSTITUTION
2018-09-28T06:03:04.8839360Z ##[debug]loaded 21
2018-09-28T06:03:05.1689206Z ##[debug]agent.proxyurl=undefined
2018-09-28T06:03:05.1689626Z ##[debug]VSTS_ARM_REST_IGNORE_SSL_ERRORS=undefined
2018-09-28T06:03:05.1689902Z ##[debug]AZURE_HTTP_USER_AGENT=VSTS_22f27331-ecdb-4f82-90a5-b10f97bd17ab_release_2_3_3_1
2018-09-28T06:03:05.2554409Z ##[debug]Agent.TempDirectory=D:\a\_temp
2018-09-28T06:03:05.2744641Z ##[debug]check path : D:\a\_tasks\AzureRmWebAppDeployment_497d490f-eea7-4f2b-ab94-48d9c1acdcb1\4.3.6\task.json
2018-09-28T06:03:05.2746968Z ##[debug]set resource file to: D:\a\_tasks\AzureRmWebAppDeployment_497d490f-eea7-4f2b-ab94-48d9c1acdcb1\4.3.6\task.json
2018-09-28T06:03:05.2747709Z ##[debug]system.culture=en-US
2018-09-28T06:03:05.2882766Z ##[debug]ConnectionType=AzureRM
2018-09-28T06:03:05.2886003Z ##[debug]WebAppKind=functionApp
2018-09-28T06:03:05.2888320Z ##[debug]DeployToSlotOrASEFlag=false
2018-09-28T06:03:05.2889163Z ##[debug]VirtualApplication=null
2018-09-28T06:03:05.2889487Z ##[debug]GenerateWebConfig=null
2018-09-28T06:03:05.2889712Z ##[debug]WebConfigParameters=null
2018-09-28T06:03:05.2891449Z ##[debug]XmlTransformation=false
2018-09-28T06:03:05.2892756Z ##[debug]JSONFiles=null
2018-09-28T06:03:05.2895720Z ##[debug]XmlVariableSubstitution=false
2018-09-28T06:03:05.2896357Z ##[debug]TakeAppOfflineFlag=false
2018-09-28T06:03:05.2898323Z ##[debug]RenameFilesFlag=false
2018-09-28T06:03:05.2898558Z ##[debug]AdditionalArguments=null
2018-09-28T06:03:05.2898728Z ##[debug]ScriptType=null
2018-09-28T06:03:05.2898910Z ##[debug]InlineScript=null
2018-09-28T06:03:05.2900854Z ##[debug]ScriptPath=D:\a\r1\a
2018-09-28T06:03:05.2901415Z ##[debug]DockerNamespace=null
2018-09-28T06:03:05.2901649Z ##[debug]AppSettings=null
2018-09-28T06:03:05.2901843Z ##[debug]StartupCommand=null
2018-09-28T06:03:05.2902019Z ##[debug]ConfigurationSettings=null
2018-09-28T06:03:05.2904100Z ##[debug]ConnectedServiceName=424ebb0f-11b3-45ba-a37e-9bddb5ee917c
2018-09-28T06:03:05.2905835Z ##[debug]WebAppName=blogfuncs
2018-09-28T06:03:05.3042520Z ##[debug]Processed: ##vso[telemetry.publish area=TaskEndpointId;feature=AzureRmWebAppDeployment]{"endpointId":"424ebb0f-11b3-45ba-a37e-9bddb5ee917c"}
2018-09-28T06:03:05.3043152Z ##[debug]Package=D:\a\r1\a\**\*.zip
2018-09-28T06:03:05.3047137Z ##[debug]Finding files matching input: D:\a\r1\a\**\*.zip
2018-09-28T06:03:05.3047387Z ##[debug]Matching glob pattern: D:\a\r1\a\**\*.zip
2018-09-28T06:03:05.3047641Z ##[debug]Index of first wildcard: 10
2018-09-28T06:03:05.3047800Z ##[debug]find root dir: D:\a\r1\a\
2018-09-28T06:03:05.3047959Z ##[debug]findPath: 'D:\a\r1\a\'
2018-09-28T06:03:05.3048114Z ##[debug]findOptions.followSpecifiedSymbolicLink: 'true'
2018-09-28T06:03:05.3048251Z ##[debug]findOptions.followSymbolicLinks: 'true'
2018-09-28T06:03:05.3048402Z ##[debug]  D:\a\r1\a\ (directory)
2018-09-28T06:03:05.3048556Z ##[debug]  D:\a\r1\a\_andrewlock.BlogFunctions (directory)
2018-09-28T06:03:05.3048699Z ##[debug]  D:\a\r1\a\_andrewlock.BlogFunctions\drop (directory)
2018-09-28T06:03:05.3048872Z ##[debug]  D:\a\r1\a\_andrewlock.BlogFunctions\drop\artifacts.zip (file)
2018-09-28T06:03:05.3049206Z ##[debug]4 results
2018-09-28T06:03:05.3049342Z ##[debug]patternRoot: ''
2018-09-28T06:03:05.3049505Z ##[debug]matchOptions.debug: 'undefined'
2018-09-28T06:03:05.3049882Z ##[debug]matchOptions.nobrace: 'undefined'
2018-09-28T06:03:05.3050027Z ##[debug]matchOptions.noglobstar: 'undefined'
2018-09-28T06:03:05.3050183Z ##[debug]matchOptions.dot: 'undefined'
2018-09-28T06:03:05.3050339Z ##[debug]matchOptions.noext: 'undefined'
2018-09-28T06:03:05.3050481Z ##[debug]matchOptions.nocase: 'true'
2018-09-28T06:03:05.3050637Z ##[debug]matchOptions.nonull: 'undefined'
2018-09-28T06:03:05.3050793Z ##[debug]matchOptions.matchBase: 'true'
2018-09-28T06:03:05.3050937Z ##[debug]matchOptions.nocomment: 'undefined'
2018-09-28T06:03:05.3051096Z ##[debug]matchOptions.nonegate: 'undefined'
2018-09-28T06:03:05.3051241Z ##[debug]matchOptions.flipNegate: 'undefined'
2018-09-28T06:03:05.3051394Z ##[debug]pattern: 'D:\a\r1\a\**\*.zip'
2018-09-28T06:03:05.3051558Z ##[debug]expanding braces
2018-09-28T06:03:05.3051700Z ##[debug]pattern: 'D:/a/r1/a/**/*.zip'
2018-09-28T06:03:05.3051864Z ##[debug]applying include pattern against original list
2018-09-28T06:03:05.3052017Z ##[debug]1 matches
2018-09-28T06:03:05.3052154Z ##[debug]1 final results
2018-09-28T06:03:05.3052304Z ##[debug]intially web config parameters :null
2018-09-28T06:03:05.3052617Z ##[debug]This is zip package 
2018-09-28T06:03:05.3052753Z ##[debug]UseWebDeploy=false
2018-09-28T06:03:05.3052904Z ##[debug]Depolyment started for windows app service
2018-09-28T06:03:05.3053057Z ##[debug]Package type of deployment is: 1
2018-09-28T06:03:05.3830783Z ##[debug]Is the package an msdeploy package : false
2018-09-28T06:03:05.3832639Z ##[debug]Predeployment Step Started
2018-09-28T06:03:05.3847828Z ##[debug]424ebb0f-11b3-45ba-a37e-9bddb5ee917c data subscriptionid = undefined
2018-09-28T06:03:05.3847999Z ##[debug]424ebb0f-11b3-45ba-a37e-9bddb5ee917c data subscriptionname = undefined
2018-09-28T06:03:05.3848973Z ##[debug]424ebb0f-11b3-45ba-a37e-9bddb5ee917c auth param serviceprincipalid = null
2018-09-28T06:03:05.3849897Z ##[debug]424ebb0f-11b3-45ba-a37e-9bddb5ee917c data environmentAuthorityUrl = undefined
2018-09-28T06:03:05.3874205Z ##[debug]Deployment Failed with Error: Error: Endpoint auth data not present: 424ebb0f-11b3-45ba-a37e-9bddb5ee917c
2018-09-28T06:03:05.3874868Z ##[debug]task result: Failed
2018-09-28T06:03:05.3944966Z ##[error]Error: Endpoint auth data not present: 424ebb0f-11b3-45ba-a37e-9bddb5ee917c
2018-09-28T06:03:05.3956259Z ##[debug]Processed: ##vso[task.issue type=error;]Error: Endpoint auth data not present: 424ebb0f-11b3-45ba-a37e-9bddb5ee917c
2018-09-28T06:03:05.3958319Z ##[debug]Processed: ##vso[task.complete result=Failed;]Error: Endpoint auth data not present: 424ebb0f-11b3-45ba-a37e-9bddb5ee917c
2018-09-28T06:03:05.3958623Z ##[debug]Deployment failed
2018-09-28T06:03:05.4015582Z ##[section]Finishing: Deploy Azure App Service
```

### Error logs
```
2018-09-28T06:03:05.3944966Z ##[error]Error: Endpoint auth data not present: 424ebb0f-11b3-45ba-a37e-9bddb5ee917c
```

**Comment by adbarlow** [comment]
I too am experiencing this error in my helm init task.

`2018-10-02T18:42:09.6724306Z ##[section]Starting: helm init
2018-10-02T18:42:09.6731160Z ==============================================================================
2018-10-02T18:42:09.6731261Z Task         : Package and deploy Helm charts
2018-10-02T18:42:09.6731570Z Description  : Deploy, configure, update your Kubernetes cluster in Azure Container Service by running helm commands.
2018-10-02T18:42:09.6731617Z Version      : 0.138.4
2018-10-02T18:42:09.6731677Z Author       : Microsoft Corporation
2018-10-02T18:42:09.6731719Z Help         : [More Information](https://go.microsoft.com/fwlink/?linkid=851275)
2018-10-02T18:42:09.6731760Z ==============================================================================
2018-10-02T18:42:10.0666853Z ##[error]Error: Endpoint auth data not present: d07c460d-9dea-4184-b262-b9dc6693d7ca
2018-10-02T18:42:10.0694998Z ##[section]Finishing: helm init
`

**Comment by chshrikh** [comment]
@adbarlow  could you please share the debug logs with us

**Comment by chshrikh** [comment]
@andrewlock is error is consistent ?  Could you please validate your service connection. You can do it by going to Project Settings > Service Connection. Click on your connection, details for connection will be displayed, there you can see "Update service connection" click on it , a new pane will be open. In that click of "use full version...."  there you can see "Verify connection" option. Click on it and check if it works fine

![image](https://user-images.githubusercontent.com/11972852/46453635-26004400-c7c0-11e8-870a-089857c2b94e.png)

![image](https://user-images.githubusercontent.com/11972852/46453697-652e9500-c7c0-11e8-8c16-442fc00aaeaf.png)

**Comment by mercenaryntx** [comment]
Hi,

I have the same issue. 
I just created a brand new project, set up the CI/CD pipelines using the built-in templates for Azure Function. The connection is verified, everything works fine, but the deployment.

My log is also full with undefined subscription details.

> 2018-10-04T10:35:02.7037887Z ##[debug]26d532ac-ae72-4c25-b6ca-d234a3481c77 data subscriptionid = undefined
2018-10-04T10:35:02.7039240Z ##[debug]26d532ac-ae72-4c25-b6ca-d234a3481c77 data subscriptionname = undefined
2018-10-04T10:35:02.7040777Z ##[debug]26d532ac-ae72-4c25-b6ca-d234a3481c77 auth param serviceprincipalid = null
2018-10-04T10:35:02.7041059Z ##[debug]26d532ac-ae72-4c25-b6ca-d234a3481c77 data environmentAuthorityUrl = undefined

**Comment by chshrikh** [comment]
@mercenaryntx can you share your contact details to RM_Customer_Queries(at)microsoft(dot)com, we can debug it further

**Comment by chshrikh** [comment]
@mercenaryntx could you please unlink the process parameters from stage. 
1) Edit your CD pipeline
2) Select you stage, in stage details you can see the option for "Unlink all" parameters and save the pipeline and do the deployment.  Ref below image.

![image](https://user-images.githubusercontent.com/11972852/46471650-261c3600-c7f8-11e8-817d-e913f95fbb04.png)

We are investigating it further, mean while please try to follow above step and check if you can get unblocked on this

**Comment by mercenaryntx** [comment]
Hi @chshrikh!

Not sure why, but that immediately solved the issue, thank you!

**Comment by chshrikh** [comment]
closing this, please follow the workaround, we are in process to fix this bug

**Comment by pierluigi** [comment]
@chshrikh Is there another issue to track the progress of the fix? Thanks!

**Comment by chshrikh** [comment]
@pierluigi we have internal one for us, will update back here once fix is available thansk

**Comment by ikewabo** [comment]
Had the same painful issue today.  After 5 hours of scratching,  I changed the Agent Pool from VS**** to Ubuntu.  This fixed it!

**Comment by kmkumaran** [comment]
@ikewabo - Can you share more details. What type of endpoint did you use, which task was used in your pipeline and the error message that you faced while using VS*** hosted pool.

**Comment by ikewabo** [comment]
I was using VS**** as  a hosted pool.  Once I changed to Ubuntu****.  It
worked for me.

**Comment by mathewjustin** [comment]
@ikewabo Same here

**Comment by dhashrath-msft** [comment]
@ikewabo @mathewjustin  Can you please share 
your organisation name or your account url 
project name and 
release / build name

where you were hitting the issue to help us investigate ?
